api/client/src/main/java/org/projectnessie/client/auth/BasicAuthenticationProvider.java

```diff
@@ -69,25 +69,13 @@ public HttpAuthentication build(Function<String, String> configSupplier) {
   private static class BasicAuthentication implements HttpAuthentication {
     private final Supplier<String> headerSupplier;
 
-    private static String basicHeader(String username, String password) {
-      if (username == null || password == null) {
-        throw new NullPointerException(
-            "username and password parameters must be present for auth type " + AUTH_TYPE_VALUE);
-      }
-
-      String userPass = username + ':' + password;
-      byte[] encoded = Base64.getEncoder().encode(userPass.getBytes(StandardCharsets.UTF_8));
-      String encodedString = new String(encoded, StandardCharsets.UTF_8);
-      return "Basic " + encodedString;
-    }
-
     private BasicAuthentication(String username, String password) {
-      String header = basicHeader(username, password);
+      String header = basicAuthorizationHeader(username, password);
```

Comment:
Why expose an auth scheme that is deprecated in Nessie in public API? Maybe add an internal utility class for it?